src/main/jbake/content/blog/2017/06/23/opentracing-appmetrics.adoc

```diff
@@ -0,0 +1,270 @@
+endif::[]
+
+Then you can select a specific trace instance and see further details:
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-06-23-trace.png[caption="Figure 3: ", title="Jaeger trace instance view"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-06-23-trace.png[caption="Figure 3: ", title="Jaeger trace instance view"]
+endif::[]
+
+This shows that the trace instance has three spans, the first representing the receipt of the `/buy` request on
+_service1_, the second where _service1_ is invoking _service2_, and finally the _service2_ receiving the
+`/hello` request. In this particular trace instance, the _service2_ invocation has reported an error, indicated
+by the `error=true` tag.
+
+Now we will look at the Grafana dashboard to see what metrics have been reported from the OpenTracing
+instrumentation within the two services:
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-06-23-grafana-dashboard.png[caption="Figure 4: ", title="Grafana dashboard"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-06-23-grafana-dashboard.png[caption="Figure 4: ", title="Grafana dashboard"]
+endif::[]
+
+This dashboard includes three graphs, the first showing the span count, the second showing the average duration
+of the spans, and third showing the ratio between successful and erronous spans.
+
+The metrics reported by Prometheus are based on a range of labels - a metric exists for each unique combination
+of those labels.
+
+The standard labels included with the OpenTracing java-metrics project are: `operation`, `span.kind` and `error`.
+
+With this particular example, we also included the `transaction` label.
+
+However when the services are deployed to Kubernetes, the following additional labels are included for free: `pod`,
+`instance`, `service`, `job` and `namespace`.
+
+In our example Prometheus queries, we have ignored most of the Kubernetes added labels (except `service`) so that
+the metrics are aggregated across the specific pods, namespaces, etc. However, having these labels available means
+it is possible to segment the metrics in whatever way is required to analyse the data.
+
+When using the `java-metrics` project outside of Kubernetes, it is still possible to include the `service`
+label, however this would be configured when setting up the tracer.
+
+We can also filter the data, to focus on specific areas of interest:
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-06-23-grafana-txn-service.png[caption="Figure 5: ", title="Customized Grafana graph focusing on metrics for transaction 'sell' and service 'service2'"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-06-23-grafana-txn-service.png[caption="Figure 5: ", title="Customized Grafana graph focusing on metrics for transaction 'sell' and service 'service2'"]
+endif::[]
+
```

Comment:
They do appear to be bold already. It's a pity the bot isn't working to generate a version of the website so you can see. Unless you have maven and want to clone the repo - and use `mvn jbake:inline` and then view the site on localhost:8080